core/utilities.lua

```diff
@@ -125,6 +125,22 @@ utilities.map = function (func, array)
   return new_array
 end
 
+utilities.sortedpairs = function (input)
+  local keys = {}
+  for k, _ in pairs(input) do
+    keys[#keys+1] = k
+  end
+  table.sort(keys, function(a, b)
+    if type(a) ~= type(b) then return false end
+    return a < b
+  end)
+  return coroutine.wrap(function()
```

Comment:
Clever and works, but I'm curious why you reached for coroutines instead of the classic Lua idiom of keeping an iteration counter outside the function.

Reply:
Well, pairs() is an iterator, and we need our own version - I just thought coroutines were[ the way you do iterators](https://www.lua.org/pil/9.3.html) in Lua. :-) (Turns out you can do it by returning a generator function as well, but truth to tell I just grabbed the structure from `utilities.gtoke`)

Reply:
Fair enough.